src/orion/core/worker/multi_task_algo.py

```diff
@@ -0,0 +1,551 @@
+    """
+    algos = {}
+    for entry_point in pkg_resources.iter_entry_points("OptimizationAlgorithm"):
+        entry_point: EntryPoint
+        algo_class = entry_point.load()
+        log.debug(
+            f"Found a {entry_point.name} from distribution: "
+            f"{entry_point.dist.project_name}={entry_point.dist.version}"
+        )
+        algos[entry_point.name] = algo_class
+    return algos
+
+
+@singledispatch
+def is_warmstarteable(
+    algo: Union[BaseAlgorithm, Type[BaseAlgorithm], Dict[str, Any]]
+) -> bool:
+    """Returns wether the given algo, algo type, or algo config, supports warm-starting.
+
+    Parameters
+    ----------
+    algo : Union[BaseAlgorithm, Type[BaseAlgorithm], Dict[str, Any]]
+        Algorithm instance, type of algorithm, or algorithm configuration dictionary.
+
+    Returns
+    -------
+    bool
+        Wether the input is or describes a warm-starteable algorithm.
+    """
+    return False
+
+
+@is_warmstarteable.register(type)
+def _(algo: Type[BaseAlgorithm]) -> bool:
+    return issubclass(algo, WarmStarteable)
+
+
+@is_warmstarteable.register(BaseAlgorithm)
+def _(algo: BaseAlgorithm) -> bool:
+    return isinstance(algo, WarmStarteable)
+
+
+@is_warmstarteable.register(AlgoWrapper)
+def _(algo: AlgoWrapper) -> bool:
+    # NOTE: Not directing directly to algo.unwrapped, because there might eventually be
+    # a Wrapper that makes an algo WarmStarteable. Recurse down the chain of wrappers
+    # instead.
+    return is_warmstarteable(algo.algorithm)
+
+
+@is_warmstarteable.register(str)
+def _(algo: str) -> bool:
+    available_algos = get_all_algos()
+    if algo not in available_algos and algo.lower() not in available_algos:
+        raise RuntimeError(
+            f"Can't tell if algo '{algo}'' is warm-starteable, since there are no "
+            f"algos registered with that name!\n"
+            f"Available algos: {list(available_algos.keys())}"
+        )
+    algo_type = available_algos.get(algo, available_algos.get(algo.lower()))
+    assert algo_type is not None
+    return inspect.isclass(algo_type) and issubclass(algo_type, WarmStarteable)
+
+
+@is_warmstarteable.register(dict)
+def _(algo: Dict[str, Any]) -> bool:
+    first_key = list(algo)[0]
+    return len(algo) == 1 and is_warmstarteable(first_key)
+
+
+class MultiTaskAlgo(AlgoWrapper):
+    """Wrapper that makes the algo "multi-task" by concatenating the task ids to the inputs.
+    """
+
+    def __init__(
+        self,
+        space: Space,
+        algorithm_config: Dict,
+        knowledge_base: AbstractKnowledgeBase,
+    ):
+        log.debug(
+            f"Creating a MultiTaskAlgo wrapper: space {space}, algo config: {algorithm_config}"
+        )
+        self.knowledge_base = knowledge_base
+        self._enabled: bool
+        if is_warmstarteable(algorithm_config):
+            log.debug(
+                "Chosen algorithm is warm-starteable, disabling this multi-task wrapper."
```

Comment:
Oh just noticed here. I believe the correct spelling is _warm-startable_. It should be fixed everywhere else as well.